src/core/mac/sub_mac.cpp

```diff
@@ -278,6 +299,32 @@ void SubMac::StartCsmaBackoff(void)
     uint32_t backoff;
     uint32_t backoffExponent = kMinBE + mTransmitRetries + mCsmaBackoffs;
 
+#if OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
+    if (mTransmitFrame.mInfo.mTxInfo.mPeriod != 0)
+    {
+        uint32_t phaseNow     = (otPlatTimeGet() / OT_US_PER_TEN_SYMBOLS) % mTransmitFrame.mInfo.mTxInfo.mPeriod;
+        uint32_t phaseDesired = mTransmitFrame.mInfo.mTxInfo.mPhase;
+
+        mTransmitFrame.SetCsmaCaEnabled(false);
+
+        SetState(kStateCslTransmit);
+
+        if (phaseNow < phaseDesired)
```

Comment:
Before the upper layer(`DataPollHandler`) requests a CSL transmission from `Mac`, it would first start a timer and make the actual request at a time a little bit ahead of the CSL phase: https://github.com/openthread/openthread/pull/4557/files#diff-e94cec5716f80ba4d7bf1c471615c1a7R414
And the max 'Ahead' time is set to **50** symbols(**8**ms). So theoratically the blocking won't be longer than **8**ms. 

However I think there's indeed a problem here: if in `StartCsmaBackoff`, the original expected time has passed and it decides to send it in next CSL period. Then this would block much time, it's related with the CSL period, which is expected to be long to save power for SSED.

What about we do the transmission immediately if the original expected time has passed(second branch)? I think this should rarely happen.

Reply:
I feel even 8ms of block may not be good. I will think more on this.

I have not read the higher layer changes in detail, but one thing that I don't understand is why `DataPollHandler` is involved in scheduling the CSL tx. `DataPollHandler` is a class for handling received data polls. 
How about adding the logic/code for CSL tx scheduling in a new class to keep it clear (e.g., `CslScheduler`, or `CslSender`)?